### PR TITLE
Remove `AddressGenerationOptions` in favor of `GenerateAddressOptions`

### LINKED ISSUE #225: Simplify AddressGenerationOptions and GenerateAddressOptions
## Description

At the moment there are two options to generate addresses available that could be simplified.

```
interface AddressGenerationOptions { // <-- Why are there two interfaces for address generation?
		internal: boolean
		options: GenerateAddressOptions 
}

interface GenerateAddressOptions {
		ledgerNanoPrompt: boolean // <-- Should this be optional? True = show prompt, false / undefined = don't show prompt 
}
```

Might be interesting to move to:

```
interface AddressGenerationOptions {
		type: AddressType.Ed25519 | AddressType.Secp256k1 // This is missing at the moment
		internal?: boolean
		ledgerNanoPrompt?: boolean 
}
```
## Motivation

Simplifies the SDK

## Are you planning to do it yourself in a pull request?

No.

### PR DESCRIPTION
**Pull request by DaughterOfMars**
# Description of change

Combines the two options structs into one.

## Links to any relevant issues

- Closes #225

## Type of change

Choose a type of change, and delete any options that are not relevant.

- Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

**Review by Thoralf-M — CHANGES_REQUESTED** [review]
Clippy and conflicts

File: sdk/src/client/api/address.rs [hunk 1]
```diff
@@ -212,8 +196,10 @@ impl<'a> GetAddressesBuilder<'a> {
                 self.coin_type,
                 self.account_index,
                 self.range,
-                true,
-                self.options.clone(),
+                self.options.map(|mut o| {
+                    o.internal = true;
+                    o
+                }),
```
Comment (Thoralf-M) [inline sdk/src/client/api/address.rs]: If options is_none() the map will not be used and it will generate public addresses

Comment (DaughterOfMars) [reply sdk/src/client/api/address.rs]: I am très stupide.

**Review by Thoralf-M — DISMISSED** [review]

**Review by Thoralf-M — DISMISSED** [review]

File: sdk/CHANGELOG.md [hunk 2]
```diff
@@ -26,6 +26,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Renamed `AccountHandle` to `Account`, `Account` to `AccountDetails` and `AccountDto` to `AccountDetailsDto`;
 - `AddressWrapper` renamed to `Bech32Address` and moved to `types`;
 - `Address::try_from_bech32_with_hrp` address and HRP return have been reversed;
+- Remove `AddressGenerationOptions` in favor of `GenerateAddressOptions`, which now contains the `internal` flag.
```
Comment (thibault-martinez) [inline sdk/CHANGELOG.md]: Can you add a `### Removed` (after `### Changed`) section and put it there please

**Review by Thoralf-M — DISMISSED** [review]

File: sdk/CHANGELOG.md [hunk 3]
```diff
@@ -30,6 +30,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - `SendNativeTokens::addresses_native_tokens` renamed to `addresses_and_native_tokens`;
 - `SendNft::addresses_nft_ids` renamed to `addresses_and_nft_ids`;
 
+### Removed
+
+- Remove `AddressGenerationOptions` in favor of `GenerateAddressOptions`, which now contains the `internal` flag.
```
Comment (thibault-martinez) [inline sdk/CHANGELOG.md]:
```suggestion
- Remove `AddressGenerationOptions` in favor of `GenerateAddressOptions`, which now contains the `internal` flag;
```

File: sdk/src/client/secret/types.rs [hunk 4]
```diff
@@ -45,13 +45,23 @@ pub struct AccountAddress {
...
+#[derive(Copy, Clone, Debug, Default, Eq, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase", default)]
 pub struct GenerateAddressOptions {
+    pub internal: bool,
     /// Display the address on ledger devices.
     pub ledger_nano_prompt: bool,
 }
 
+impl GenerateAddressOptions {
+    pub const fn internal() -> Self {
```
Comment (kwek20) [inline sdk/src/client/secret/types.rs:57]: we may want this in bindings as well, its quite convenient :)

File: sdk/tests/client/addresses.rs [hunk 5]
```diff
@@ -173,7 +170,10 @@ async fn address_generation() {
             .with_coin_type(address.coin_type)
             .with_account_index(address.account_index)
             .with_range(address.address_index..address.address_index + 1)
-            .with_internal_addresses(address.internal)
+            .with_options(GenerateAddressOptions {
```
Comment (kwek20) [inline sdk/tests/client/addresses.rs:173]: We could do this using a GenerateAddressOptions ::from(AddressData) or something, but thats not much prettier i guess

File: sdk/tests/wallet/mod.rs [hunk 6]
```diff
@@ -12,6 +12,7 @@ mod claim_outputs;
 mod common;
 mod consolidation;
 mod error;
+#[cfg(feature = "message_interface")]
```
Comment (kwek20) [inline sdk/tests/wallet/mod.rs:15]: much cleaner 🙏

File: sdk/tests/client/addresses.rs (hunk 5 above)
Comment (kwek20) [reply sdk/tests/client/addresses.rs:173]: yea never mind, would only be usefull to clean this test, thats useless then

**Review by kwek20 — APPROVED** [review]

**Review by Alex6323 — APPROVED** [review]

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**